modulus/models/afno/modafno.py

```diff
@@ -0,0 +1,649 @@
+            )
+
+        self.norm2 = norm_layer(embed_dim)
+        mlp_latent_dim = int(embed_dim * mlp_ratio)
+        if modulate_mlp:
+            self.mlp = ModAFNOMlp(
+                in_features=embed_dim,
+                latent_features=mlp_latent_dim,
+                out_features=embed_dim,
+                mod_features=mod_dim,
+                activation_fn=activation_fn,
+                drop=drop,
+            )
+        else:
+            self.mlp = AFNOMlp(
+                in_features=embed_dim,
+                latent_features=mlp_latent_dim,
+                out_features=embed_dim,
+                activation_fn=activation_fn,
+                drop=drop,
+            )
+        self.double_skip = double_skip
+        self.modulate_filter = modulate_filter
+        self.modulate_mlp = modulate_mlp
+
+    def forward(self, x: Tensor, mod_embed: Tensor) -> Tensor:
+        residual = x
+        x = self.norm1(x)
+        x = self.filter(x, mod_embed) if self.modulate_filter else self.filter(x)
+
+        if self.double_skip:
+            x = x + residual
+            residual = x
+
+        x = self.norm2(x)
+        x = self.mlp(x, mod_embed) if self.modulate_mlp else self.mlp(x)
+        x = x + residual
+        return x
+
+
+@dataclass
+class MetaData(ModelMetaData):
+    name: str = "ModAFNO"
+    # Optimization
+    jit: bool = False  # ONNX Ops Conflict
+    cuda_graphs: bool = True
+    amp: bool = True
+    # Inference
+    onnx_cpu: bool = False  # No FFT op on CPU
+    onnx_gpu: bool = True
+    onnx_runtime: bool = True
+    # Physics informed
+    var_dim: int = 1
+    func_torch: bool = False
+    auto_grad: bool = False
+
+
+class ModAFNO(Module):
+    """Modulated Adaptive Fourier neural operator (ModAFNO) model.
+
+    Parameters
+    ----------
+    inp_shape : List[int]
+        Input image dimensions [height, width]
+    in_channels : int, optional
+        Number of input channels
+    out_channels: int, optional
+        Number of output channels
+    embed_model: dict, optional
+        Dictionary of arguments to pass to the `ModEmbedNet` embedding model
+    patch_size : List[int], optional
+        Size of image patches, by default [16, 16]
+    embed_dim : int, optional
+        Embedded channel size, by default 256
+    depth : int, optional
+        Number of AFNO layers, by default 4
+    mlp_ratio : float, optional
+        Ratio of layer MLP latent variable size to input feature size, by default 4.0
+    drop_rate : float, optional
+        Drop out rate in layer MLPs, by default 0.0
+    num_blocks : int, optional
+        Number of blocks in the block-diag frequency weight matrices, by default 16
+    sparsity_threshold : float, optional
+        Sparsity threshold (softshrink) of spectral features, by default 0.01
+    hard_thresholding_fraction : float, optional
+        Threshold for limiting number of modes used [0,1], by default 1
+    modulate_filter: bool, optional
```

Comment:
Nit: add `, by default True` and order these the same as the parameters of the constructor.

Reply:
Fixed in https://github.com/NVIDIA/modulus/pull/547/commits/0b75c9b0348c95a3096ead5437d2222dc18f4418.